revista-lecciones-y-ensayos.csl

```diff
@@ -0,0 +1,420 @@
+  <macro name="author">
+    <group>
+      <names variable="author" font-variant="normal">
+        <name font-variant="normal" and="symbol" delimiter-precedes-et-al="never" delimiter-precedes-last="never" initialize="false" initialize-with="." name-as-sort-order="all">
+          <name-part name="family" font-variant="small-caps"/>
+        </name>
+        <label form="short" text-case="lowercase" strip-periods="true" font-variant="normal" prefix=" "/>
+        <substitute>
+          <names variable="editor"/>
+        </substitute>
+      </names>
+    </group>
+  </macro>
+  <macro name="author-short">
+    <names variable="author">
+      <name form="short" and="symbol" delimiter-precedes-et-al="never" delimiter-precedes-last="never" et-al-use-first="3" initialize-with=".">
+        <name-part name="family" font-variant="small-caps"/>
+      </name>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="container-author">
+    <group>
+      <names variable="container-author" font-variant="normal">
+        <name font-variant="normal" delimiter=", " and="symbol" delimiter-precedes-last="always" initialize="false" initialize-with="" name-as-sort-order="all">
+          <name-part name="family" font-variant="small-caps"/>
+        </name>
+        <label form="short" text-case="lowercase" strip-periods="true" font-variant="normal" prefix=" "/>
+      </names>
+    </group>
+  </macro>
+  <macro name="translator">
+    <group>
+      <names variable="translator" font-variant="normal">
+        <name font-variant="normal" delimiter=", " and="symbol" delimiter-precedes-last="always" initialize="false" initialize-with="" name-as-sort-order="all">
+          <name-part name="family" font-variant="small-caps"/>
+        </name>
+        <label form="short" text-case="lowercase" prefix=" (" suffix=")"/>
+      </names>
+    </group>
+  </macro>
+  <macro name="title">
+    <text variable="title" quotes="false" font-style="italic" text-decoration="none"/>
+  </macro>
+  <macro name="locator">
+    <label strip-periods="true" suffix="." variable="locator" form="short"/>
+    <text variable="locator" prefix=" "/>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=": ">
+      <text variable="publisher" prefix=" "/>
+    </group>
+  </macro>
+  <macro name="edition">
+    <choose>
+      <if is-numeric="edition">
+        <group delimiter=" ">
+          <number variable="edition"/>
+          <text value="ed."/>
+        </group>
+      </if>
+      <else>
+        <text variable="edition" suffix="."/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="year-issued">
+    <date variable="issued">
+      <date-part name="year"/>
+    </date>
+  </macro>
+  <citation et-al-min="4" et-al-use-first="3" disambiguate-add-year-suffix="true" collapse="year-suffix">
+    <layout delimiter="; ">
+      <choose>
+        <if type="book">
```

Comment:
Please change subsequent item types to use "else-if" with the same "choose", rather than starting a new "choose" block for each. 

Please add an "else" at the end to capture other item types you haven't explicitly added, like article-magazine and webpage